cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -162,20 +165,48 @@ private void createCruiseControlMetricsTopic() {
       createTopicsResult.values().get(_newTopic.name()).get();
       LOG.info("Cruise Control metrics topic created: {}", _cruiseControlMetricsTopic);
     } catch (InterruptedException | ExecutionException e) {
-      if (!(e.getCause() instanceof TopicExistsException)) {
-        LOG.error("Unable to create Cruise Control topic", e);
-      }
+      LOG.warn("Unable to create Cruise Control metrics topic " + _cruiseControlMetricsTopic, e);
```

Comment:
We have some common logic regarding (1) creating non-existing topic and (2) modifying the selected configs of existing topics (i.e. here and in sample store topic creation).
Can we make the relevant logic (1) shared and (2) pluggable between the two? The first part will ensure a common code path for similar execution. The second part will enable users to add their custom logic that they may want to automate during topic creation (e.g. generating ACLs).

Reply:
I was thinking about this.  
One concern prevent me doing that is where should I put the common code. Because `cruise-control` and `cruise-control-metrics-reporter` are two separate module and we do not want one to depends on the other. And it also seems inappropriate to put this code in `cruise-control-core` module since it only contains non-kafka-specific code.

Reply:
Sure, I agree with your concerns regarding sharing the logic.
Are there any issues with making each logic pluggable within their corresponding modules?

Reply:
While I was taking a deeper look, I just realize that `CruiseControlMetricsReporter` and `KafkaSampleStore` themselves are `Configurable` class, then to make the topic management logic pluggable, we can simply change corresponding methods to `protected`.

For example, if someone want to add some more check for metrics topic, (s)he can directly extends from `CruiseControlMetricsReporter` and override `createNewTopicFromReporterConfig()`, `createCruiseControlMetricsTopic()` and `maybeUpdateCruiseControlMetricsTopic()` methods.

and then change the config `metric.reporters` to the extended class name.